### DISCUSSION TITLE
Yolov8 / Deepsort re-identification advice

### DISCUSSION THREAD

**Discussion by Slipstream84** [ultralytics/ultralytics#15886] [Q&A]
Hey everyone,

I have a situation where I would like to identify / track boats leaving and returning (several hours later) from a fixed location, with the idea being to estimate how long they've been away for.
From my very limited knowledge and research, I ended up here using Yolov8 and Deepsort as a tracker.

I've got the detections to work pretty well and the tracker works ok sometimes, but in my research as to why it's been patchy I've realized the test videos are a source of issue - They're highly time-lapsed. 1 frame is 7.2 seconds, so when an object moves faster than a snail, the bounding box information and IoU information are too far out and a new ID is generated.
I've edited the nn_budget and IoU allowance for Deepsort and had some limited success in retaining correct IDs for moving targets, but once an object leaves the frame for 2 hours (1000 frames), it gives it a new ID.

I know these tracking models are built and coded more for "streaming" video, and the videos I've been given represent more a stream of separate images, than a video stream.

I'm looking for advice on a way forward to achieving my goal - particular areas of general code I should be looking at, or image re-identification models that I should research, that sort of thing or something else that I'm new enough not to even know to ask about
I'm going to ask for better videos, but I don't think that'll eventuate.

Generally I like working things out myself and not asking for help, but there's a deadline that steadily approaching and I'm new enough to this field that I'm overwhelmed with information.

Any help would be greatly appreciated =)
Cheers,
--Nick

**Comment by glenn-jocher** [comment]
@Slipstream84 hi Nick,

For your scenario, consider exploring re-identification models specifically designed for long-term tracking, which might handle the time-lapse issue better. Additionally, adjusting the Deepsort parameters further or integrating a custom re-ID model could improve ID retention. If possible, obtaining videos with a higher frame rate would also be beneficial. For more detailed guidance, you can refer to our documentation: https://docs.ultralytics.com/modes/track/. Good luck!
